control_panel_api/services.py

```diff
@@ -199,45 +206,29 @@ def delete_bucket_policies(bucket_name):
     aws.delete_policy(policy_arn_readonly)
 
 
-def detach_bucket_access_from_role(bucket_name, readwrite, role_name):
-    policy_arn = _policy_arn(
-        bucket_name=bucket_name,
-        readwrite=readwrite
-    )
-
-    aws.detach_policy_from_role(
-        policy_arn=policy_arn,
-        role_name=role_name
+@contextmanager
+def s3_access_policy(role_name):
+    policy_document = aws.get_inline_policy_document(
+        role_name=role_name,
+        policy_name=S3_POLICY_NAME,
     )
+    policy = S3AccessPolicy(document=policy_document)
 
+    yield policy
 
-def attach_bucket_access_to_role(bucket_name, readwrite, role_name):
-    policy_arn = _policy_arn(
-        bucket_name,
-        readwrite,
-    )
-
-    aws.attach_policy_to_role(
-        policy_arn=policy_arn,
+    aws.put_role_policy(
         role_name=role_name,
+        policy_name=S3_POLICY_NAME,
+        policy_document=policy.document,
     )
 
 
-def update_bucket_access(bucket_name, readwrite, role_name):
-    new_policy_arn = _policy_arn(
-        bucket_name,
-        readwrite,
-    )
-    old_policy_arn = _policy_arn(
-        bucket_name,
-        not readwrite,
-    )
+def revoke_bucket_access(bucket_arn, role_name):
+    with s3_access_policy(role_name) as policy:
```

Comment:
I don't think this is appropriate use of a context manager.  A CM is usually used to provide extra functionality such as automatically closing a file once you've exited the block of code.

From what I can see in your CM it's just calling a function.  The use of a CM provides you no benefit, unless I'm missing something.  This code IMO should just be a regular call to a function.  

```
policy = get_s3_access_policy(role_name)
```

Reply:
The context manager reads the policy from AWS, yields it to the calling code to modify as needed, then writes the modified policy back to AWS. It encapsulates dealing with AWS, leaving the calling code to worry only about the changes to the policy. Pretty much the definition of managing context, in my opinion.

Reply:
Ohhh I didn't see the PUT in the diff.  Thanks for explanation.  @xoen ignore me.

Reply:
No problem 👍